tests/metrics/test_compare_ground_truth.py

```diff
@@ -0,0 +1,289 @@
+    generate_pred(start_index=500, end_index=522, label="State", text="B"),
+    generate_pred(start_index=100, end_index=106, label="Zip", text="A"),
+    generate_pred(start_index=200, end_index=206, label="Zip", text="B"),
+    generate_pred(start_index=300, end_index=306, label="Zip", text="C"),
+    generate_pred(start_index=400, end_index=406, label="Zip", text="D"),
+    generate_pred(start_index=500, end_index=506, label="Zip", text="E"),
+    generate_pred(start_index=600, end_index=606, label="Zip", text="F"),
+    generate_pred(start_index=700, end_index=706, label="Zip", text="G"),
+    generate_pred(start_index=800, end_index=806, label="Zip", text="H"),
+    generate_pred(start_index=900, end_index=906, label="Zip", text="H"),
+]
+expected_labels = [
+    "Address",
+    "Amount",
+    "Business Category",
+    "City",
+    "Date",
+    "PO Number",
+    "State",
+    "Vendor Name",
+    "Zip",
+]
+expected_all_label_metrics = {
+    "Address": {
+        "false_negatives": 0,
+        "false_positives": 0,
+        "precision": 1.0,
+        "recall": 1.0,
+        "true_positives": 3,
+    },
+    "Amount": {
+        "false_negatives": 2,
+        "false_positives": 1,
+        "precision": 0.6666666666666666,
+        "recall": 0.5,
+        "true_positives": 2,
+    },
+    "Business Category": {
+        "false_negatives": 3,
+        "false_positives": 4,
+        "precision": 0,
+        "recall": 0,
+        "true_positives": 0,
+    },
+    "City": {
+        "false_negatives": 3,
+        "false_positives": 0,
+        "precision": 0,
+        "recall": 0,
+        "true_positives": 0,
+    },
+    "Date": {
+        "false_negatives": 1,
+        "false_positives": 1,
+        "precision": 0.75,
+        "recall": 0.75,
+        "true_positives": 3,
+    },
+    "PO Number": {
+        "false_negatives": 0,
+        "false_positives": 4,
+        "precision": 0,
+        "recall": 0,
+        "true_positives": 0,
+    },
+    "State": {
+        "false_negatives": 1,
+        "false_positives": 0,
+        "precision": 1.0,
+        "recall": 0.6666666666666666,
+        "true_positives": 2,
+    },
+    "Vendor Name": {
+        "false_negatives": 2,
+        "false_positives": 1,
+        "precision": 0.5,
+        "recall": 0.3333333333333333,
+        "true_positives": 1,
+    },
+    "Zip": {
+        "false_negatives": 0,
+        "false_positives": 3,
+        "precision": 0.6666666666666666,
+        "recall": 1.0,
+        "true_positives": 6,
+    },
+}
+expected_overall_metrics = {
+    "true_positives": 17,
+    "false_positives": 14,
+    "false_negatives": 12,
+    "precision": (17 / (17 + 14)),
+    "recall": (17 / (17 + 12)),
+}
+
+
+@pytest.fixture(scope="function")
+def ex_cgt_object():
+    cgt_object = CompareGroundTruth(ground_truth_list, predictions_list)
+    cgt_object.set_all_label_metrics("overlap")
+    cgt_object.set_overall_metrics()
```

Comment:
generally- we shouldn't use our fixture to call all of the methods we want to be testing. In this instantiating the class doesn't require a lot of processing, so don't need to put it in a fixture at all.